src/sidechain.cpp

```diff
@@ -171,4 +179,116 @@ void UpdateDrivechains(const CTransaction& tx, CCoinsViewCache& view, CTxUndo &t
     proposal_list << withdraw_proposal_list;
     COutPoint record_id{ArithToUint256(arith_uint256{uint64_t{block_height}}), DBIDX_SIDECHAIN_PROPOSAL_LIST};
     CreateDBEntry(view, txundo, block_height, record_id, MakeByteSpan(proposal_list));
+
+    // TODO: clean up expiring stuff
+    // TODO: activate sidechains and assign CTIPs
+}
+
+bool VerifyDrivechainSpend(const CTransaction& tx, const unsigned int sidechain_input_n, const std::vector<CTxOut>& spent_outputs, const CCoinsViewCache& view, TxValidationState& state) {
```

Comment:
It seems here that one has to spend the CTIP in order to deposit to a sidechain. And if there are multiple deposits in a block then each deposit after the first one must spend the previous unconfirmed CTIP in order to be accepted.

If the fee rate required to be included in a block is higher than the minimum fee to be included in miner mempools, this seems to open the door for "freeloaders" who pay the minimum fee to be included in the mempool, hoping to be subsidised by subsequent depositors within the same block - as subsequent depositors are forced to spend the freeloader's unconfirmed CTIP in order to be accepted into mempools. This seems like it could result in a bad user experience.

Is it not possible to make it so that deposits do not have to spend the CTIP, but rather make an additional output to the sidechain address? Then, when a block is mined, there can be a different rule where the miner is forced to make a consolidation transaction merging all the sidechain UTXOs into one. So essentially, the "only one CTIP allowed" rule is only checked based on the final state once all transactions in the block are accounted for.

I'm not sure how this would affect the overall size of the deposits, but it seems that it would be minimal. There'd be 1 more transaction, 1 more input and 1 more output in total compared to the current method. We could also allow both methods of depositing and users can choose based on the circumstances.

Reply:
I believe the end goal, if Bitcoin moves forward with this, is to have miners rewrite deposits on the fly to update the CTIP as appropriate.